pkg/catalog/catalog.go

```diff
@@ -1833,7 +1844,8 @@ func (c *Catalog) writeUncommittedLocal(ctx context.Context, repository *gravele
 	}
 
 	// Finished reading all staging area - no continuation token
-	return nil, hasData, nil
+	hasMore := count > 0
```

Comment:
Suggest to change the param name back to hasData - this is an indication to whether we need to write anything to parquet, not if there's more data to read